server/src/main/java/io/crate/types/DataType.java

```diff
@@ -82,6 +83,17 @@
 
     public abstract Streamer<T> streamer();
 
+    /**
+     * Must be used only in the cast functions.
+     *
+     * @param value The value to cast to the target {@link DataType}.
+     * @param mode  The {@link CastMode} mode.
+     * @return The value of the target {@link DataType}.
+     */
+    public T cast(Object value, CastMode mode) throws IllegalArgumentException, ClassCastException {
```

Comment:
probably `cast(Object value, boolean explicitCast)` should be ok?

Reply:
Would also be okay. Or a new method? `explicitCast` and `implicitCast` ? `explicitCast` could default to `implicitCast` if there is no additional logic.